lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -151,6 +150,11 @@ sub job_grab {
     my $workercaps = $args{workercaps};
 
     my $worker = _validate_workerid($workerid);
+    if ($worker->job) {
+        # TROUBLE ahead - do not call seen, to get into dead worker detection first
```

Comment:
I'm sorry, I still do not understand. What is harmful on calling `$worker->seen(..)` and doing this check after?